src/modules/LanUdevAuto.rb

```diff
@@ -194,39 +198,34 @@ def Import(settings)
     end
 
     def Write
-      comment = [
-        "# Generated by autoyast",
-        "# program run by the persistent-net-generator.rules rules file.",
-        "#",
-        "# You can modify it, as long as you keep each rule on a single line."
-      ]
+      template = "SUBSYSTEM==\"net\", ACTION==\"add\", DRIVERS==\"?*\", %s==\"%s\", NAME=\"%s\""
+
       rules = []
-      Builtins.foreach(@udev_rules) do |rule|
-        rules = Builtins.add(
-          rules,
-          Builtins.sformat(
-            "SUBSYSTEM==\"net\", ACTION==\"add\", DRIVERS==\"?*\", %1==\"%2\", NAME=\"%3\"",
-            Ops.get_string(rule, "rule", ""),
-            Ops.get_string(rule, "value", ""),
-            Ops.get_string(rule, "name", "")
-          )
-        )
+
+      @udev_rules.each do |rule|
+        opt = rule["rule"] || ""
+        value = rule["value"] || ""
+        devname = rule["name"] || ""
+
+        rules << template % [opt, value.downcase, devname]
       end
```

Comment:
in fact instead of each you should use map ( so `roles = []` is not needed )
like:
```
     rules = @udev_rules.map do |rule|
        opt = rule["rule"] || ""
        value = rule["value"] || ""
        devname = rule["name"] || ""

        template % [opt, value.downcase, devname]
     end
```